theseus/optimizer/nonlinear/nonlinear_optimizer.py

```diff
@@ -262,8 +262,11 @@ def _optimize_loop(
                 step_size = self.params.step_size
                 force_update = False
 
-            self.retract_and_update_variables(
-                delta, converged_indices, step_size, force_update=force_update
+            self.objective.step_optim_vars(
+                delta * step_size,
```

Comment:
Maybe worth adding an explicit line that does `delta *= step_size` before this function call?

Reply:
Is this necessary? I don't see delta being used anywhere else after this function call.

Reply:
Probably okay, just wondering if there is a chance of introducing bugs in the future by having a variable change in a function call.